File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlNullValue.cs
namespace T1.SqlSharp.Expressions;

public class SqlNullValue : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.Null;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_NullValue(this);
    }

    public string ToSql()
    {
        return "NULL";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlNullValue.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlNullValue implements ISqlExpression {
    SqlType: SqlType = SqlType.Null;
    Span: TextSpan = new TextSpan();

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_NullValue(this);
    }

    ToSql(): string {
        return 'NULL';
    }
} 